lib/response.js

```diff
@@ -61,12 +61,12 @@ module.exports = res
 
 res.status = function status(code) {
   // Check if the status code is not an integer
-  if (!Number.isInteger(code)) {
-    throw new TypeError(`Invalid status code: ${JSON.stringify(code)}. Status code must be an integer.`);
+  if (typeof code !== 'number' || !Number.isInteger(code)) {
+    throw new TypeError(`Expected an integer HTTP status code, but received: ${JSON.stringify(code)}`);
   }
   // Check if the status code is outside of Node's valid range
   if (code < 100 || code > 999) {
-    throw new RangeError(`Invalid status code: ${JSON.stringify(code)}. Status code must be greater than 99 and less than 1000.`);
+    throw new RangeError(`Invalid status code: ${JSON.stringify(code)}. Status code must be between 99 and 999.`);
```

Comment:
This change makes the message less clear and wrong, regardless of interpretation:

- greater than 99 and less than 1000 => `99 < code < 1000` (correct)
- between 99 and 999 could be interpreted as:
  - (99, 999) => `99 < code < 999` (wrong)
  - [99, 999] => `99 <= code <= 999` (wrong)
  - [99, 999) => `99 <= code < 999` (wrong)